src/data_types.js

```diff
@@ -156,33 +156,43 @@ class BOOLEAN extends DataType {
   }
 }
 
+const LENGTH_VARIANTS = [ 'tiny', '', 'medium', 'long' ];
+
 class TEXT extends DataType {
-  constructor() {
+  constructor(length = '') {
+    if (!LENGTH_VARIANTS.includes(length)) {
+      throw new Error(`invalid blob length: ${length}`);
+    }
     super();
     this.dataType = 'text';
+    this.length = length;
   }
 
   toSqlString() {
-    return this.dataType.toUpperCase();
+    return [ this.length, this.dataType ].join('').toUpperCase();
   }
 }
 
 class BLOB extends DataType {
-  constructor() {
+  constructor(length = '') {
+    if (!LENGTH_VARIANTS.includes(length)) {
+      throw new Error(`invalid blob length: ${length}`);
+    }
     super();
-    this.dataType = 'longblob';
+    this.dataType = 'blob';
+    this.length = length;
   }
 
   toSqlString() {
-    return this.dataType.toUpperCase();
+    return [ this.length, this.dataType ].join('').toUpperCase();
```

Comment:
I'd say mysql is crazy about types. Again, TINYBLOB, BLOB, MEDIUMBLOB, and LONGBLOB